lifecycle/lifecycle/server/metrics.py

```diff
@@ -67,15 +91,26 @@ def collect_active_threads_metric() -> Iterator[Metric]:
 
 
 def collect_tcp_connections_metric() -> Iterator[Metric]:
+    net_connections = psutil.net_connections(kind='tcp')
+    tcp_connections = collections.Counter(conn.status for conn in net_connections)
     metric_name = 'lifecycle_tcp_connections_count'
-    tcp_connections = collections.Counter(p.status for p in psutil.net_connections(kind='tcp'))
     prometheus_metric = GaugeMetricFamily(metric_name, 'Number of open TCP connections by status')
     for status, count in tcp_connections.items():
         prometheus_metric.add_sample(metric_name, {
             'status': status,
         }, count)
     yield prometheus_metric
 
+    remote_ports = [c.raddr.port for c in net_connections if c.status == 'ESTABLISHED' and c.raddr.port < 10000]
```

Comment:
What ports are above 10000 that we don't want metrics for?

Also the '10000' could be a variable as we use it below too.

Reply:
That's a good question. Initially, I didn't have that limit, but then I started seeing some random ports. Perhaps, it's connecting to its own local services. I was worried about Prometheus metrics exploding in its cardinality, that's why I've set this limit.
I think it's reasonable, cause I guess there is some sort of a common sense that the important apps use 4 digit ports.